### PR TITLE
PARQUET-2243: Support zstd-jni in DirectCodecFactory

### PR DESCRIPTION
**Pull request by gszadovszky**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-2243
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Enabled executing `parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestDirectCodecFactory` for ZSTD encoding
### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/example/TestInputOutputFormat.java [hunk 1]
```diff
@@ -66,7 +66,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class TestInputOutputFormat {
+public class
+TestInputOutputFormat {
```
Comment (wgtmac) [inline parquet-hadoop/src/test/java/org/apache/parquet/hadoop/example/TestInputOutputFormat.java:70]: Modified by accident?

**Review by wgtmac — APPROVED** [review]

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/example/TestInputOutputFormat.java (hunk 1 above)
Comment (gszadovszky) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/example/TestInputOutputFormat.java:70]: Yes, thanks for the noticing.

Comment (wgtmac) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/example/TestInputOutputFormat.java]: Looking good on my side. Thanks @gszadovszky

**Comment by gszadovszky** [comment]
Thank you @wgtmac for the review!
I'll push it tomorrow if are no objections.

**Pull request merged**